.github/workflows/cd.yml

```diff
@@ -23,12 +23,31 @@ jobs:
     with:
       image_version: ${{ github.sha }}
 
-  functional-tests-local:
+  set-component-versions:
+    needs: publish-container-image
+    runs-on: ubuntu-latest
+    outputs:
+      worker_version: ${{ steps.worker_version.outputs.VERSION }}
+      cli_version: ${{ steps.cli_version.outputs.VERSION }}
+
+    steps:
+      - id: worker_version
+        name: dynamic input worker version
+        run: |
+          if [[ "${{ inputs.worker_version }}" == "" ]]; then echo "VERSION=latest" >> $GITHUB_OUTPUT; else echo "VERSION=${{ inputs.worker_version }}" >> $GITHUB_OUTPUT;fi
```

Comment:
Kairo, can you please remind me: why we write `VERSION=latest` in the `GITHUB_OUTPUT` variable instead directly writing `latest`?
I am asking as then on lines 49 and 50 we get `needs.set-component-versions.outputs.worker_version` and `needs.set-component-versions.outputs.cli_version` which will get the output of the `set-component-versions` job and more precisely from `worker_version` and `cli_version` steps respectively.

Reply:
This is how GitHub Actions works
https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs#example-defining-outputs-for-a-job